sympy/solvers/tests/test_solveset.py

```diff
@@ -1749,3 +1749,58 @@ def test_issue_14454():
     number = CRootOf(x**4 + x - 1, 2)
     raises(ValueError, lambda: invert_real(number, 0, x, S.Reals))
     assert invert_real(x**2, number, x, S.Reals)  # no error
+
+
+def test_transolve():
+    from sympy.abc import x, y
+    from sympy.simplify import simplify
+
+    #### exponential equations ####
+
+    e1 = 3**(2*x) - 2**(x + 3)
+    e2 = 4**(5 - 9*x) - 8**(2 - x)
+    e3 = 2**x + 4**x
+    e4 = exp(log(5)*x) - 2**x
+    e5 = x**(y*z) - x     # issue 10864
+    e6 = 5**(x/2) - 2**(x/3)
+    e7 = 2**(x) + 4**(x) + 8**(x) - 84
+    e8 = 4**(x+1) + 4**(x+2) + 4**(x-1) - 3**(x+2) -3**(x+3)
+
+    f1 = exp(x/y)*exp(-z/y) - 2
+    f2 = x**x
+
+    assert solveset(e1, x, S.Reals) == FiniteSet(-3*log(2)/(-2*log(3) + log(2)))
+    assert simplify(solveset(e2, x, S.Reals)) == FiniteSet(4/S(15))
```

Comment:
Need to return a simplified result for such equations.
eg: 2**x - 32 currently gives `{log(32)/log(2)}`, should rather be {5}.
Is it a good idea to use `simplify()` [here](https://github.com/sympy/sympy/blob/master/sympy/solvers/solveset.py#L930)?

Reply:
Raw `simplify` can be a very expensive function call and is generally avoided in library code. In the old `solve` there is a flag to enable or disable `simplify`. 

Is it possible instead to invert powers using a two argument log, like `log(32, 2)`?

Reply:
@asmeurer it seems that two argument log does not work well with `imageset`. I tried

```
>>> g = FiniteSet(32)
>>> imageset(Lambda(n, log(n, 2)), g)
{log(32)/log(2)}
```

Reply:
opened #14746 for this. Probably should solve this problem.